controllers/managedocs_controller.go

```diff
@@ -551,25 +572,73 @@ func (r *ManagedOCSReconciler) reconcileAlertmanager() error {
 	return nil
 }
 
-func (r *ManagedOCSReconciler) reconcileAlertmanagerConfig() error {
-	// r.Log.Info("Reconciling AlertmanagerConfig")
-	// _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.alertmanagerConfig, func() error {
-	// 	if err := r.own(r.alertmanagerConfig); err != nil {
-	// 		return err
-	// 	}
+func (r *ManagedOCSReconciler) reconcileAlertmanagerConfigSecret() error {
+	r.Log.Info("Reconciling AlertmanagerConfig secret")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.alertmanagerConfigSecret, func() error {
+		if err := r.own(r.alertmanagerConfigSecret); err != nil {
+			return err
+		}
+
+		var pagerdutyServiceKey string
 
-	// 	desired := utils.ObjectFromTemplate(templates.AlertmanagerConfigTemplate, r.Scheme).(*promv1a1.AlertmanagerConfig)
-	// 	r.alertmanagerConfig.ObjectMeta.Labels = map[string]string{monLabelKey: monLabelValue}
-	// 	// TODO: Setup pagerduty and deadman sqtich on the desired alertmanagr config
-	// 	r.alertmanagerConfig.Spec = desired.Spec
+		if err := r.get(r.pagerdutySecret); err != nil {
+			return fmt.Errorf("Unable to get pagerduty secret: %v", err)
+		}
 
-	// 	return nil
-	// })
-	// if err != nil {
-	// 	return err
-	// }
+		pagerdutySecretData := r.pagerdutySecret.Data
+		if len(pagerdutySecretData["PAGERDUTY_KEY"]) != 0 {
+			pagerdutyServiceKey = string(pagerdutySecretData["PAGERDUTY_KEY"])
+		}
+		if len(pagerdutySecretData["serviceKey"]) != 0 {
+			pagerdutyServiceKey = string(pagerdutySecretData["serviceKey"])
+		}
+		if pagerdutyServiceKey == "" {
+			return fmt.Errorf("Pagerduty service key is not defined")
+		}
```

Comment:
[suggestion] This code block can be written in a simpler way:
```suggestion
		pagerdutyServiceKey := pagerdutySecretData["PAGERDUTY_KEY"]
		if pagerdutyServiceKey == "" {
                         pagerdutyServiceKey = pagerdutySecretData["serviceKey"]
		}
		if pagerdutyServiceKey == ""  {
			return fmt.Errorf("Pagerduty service key is not defined")
		}
```

In any case, please do use one form of checking that a string is empty, either `s == ""` or `len(s) == 0` but do not switch between the to, as it makes the code more convoluted

Reply:
In addition, I think the error message is not specific enough, I will suggest something like: 
"Pagerduty secret does not contain a serviceKey or PAGERDUTY_KEY entries"